pkg/server/catalog/catalog.go

```diff
@@ -202,11 +273,8 @@ func loadSQLDataStore(ctx context.Context, config Config, coreConfig catalog.Cor
 
 	dsLog := config.Log.WithField(telemetry.SubsystemName, sqlConfig.Name)
 	ds := ds_sql.New(dsLog)
-	configurer := catalog.ConfigurerFunc(func(ctx context.Context, _ catalog.CoreConfig, configuration string) error {
-		return ds.Configure(ctx, configuration)
-	})
-
-	if _, err := catalog.ConfigurePlugin(ctx, coreConfig, configurer, sqlConfig.DataSource, ""); err != nil {
+	dsConfigurer := &dsConfigurer{ds: ds}
```

Comment:
NIT:  to avoid using the struct name as variable can you update var name?
```suggestion
	dsConf := &dsConfigurer{ds: ds}
```